### DISCUSSION TITLE
River 3 Support

### DISCUSSION THREAD

**Discussion by aad4321** [tolwi/hassio-ecoflow-cloud#374] [Ideas]
River 3 is an ultra popular device being sold by Ecoflow and would be a wonderful addition to the app.

**Comment by tuxevil** [comment]
+1 to this
I have RIVER 2, RIVER 2 MAX and I just got RIVER 3
Please add support to it

**Comment by P3TACCO** [comment]
River 3 Plus too please.

Tried River 2 ( the serial number for my River 3 Plus is in my Ecoflow account, the integration created a device, but the comms don't appear to be working)

Thanks

**Comment by P3TACCO** [comment]
I copied the River 2 Pro device files (one in internal, the other in public)
I renamed them river3_plus and changed the code inside anywhere it referenced River2Pro to as River3Plus
I uploaded those files into HA.

I modified the device-registry.py file to include the new devices and rebooted.

I now have a River 3 Plus device I can add, and it allows me to add it with my River 3 Plus serial number.

But it doesn't appear to be working - the entity values are all zero.

Need to look into the API - the call names may have changed for the River 3 devices.

**Comment by P3TACCO** [comment]
My Delta 2s work
Sample Battery sensor:
Main Battery Level (bms_bmsStatus.soc)

In the River 2 Pro file it's the same
Main Battery Level (bms_bmsStatus.soc)

But that's not working in my replicated River 3

**Comment by P3TACCO** [comment]
For Instance

The DC Out commands are different for Delta and River

Delta
DC Out Power (mppt.outWatts)

River
DC Out Power (pd.carWatts)

**Comment by P3TACCO** [comment]
Looks like my Delta 2s are no longer working either.  I suspect this is an Ecoflow server issue. 

Rollback to 1.1.0 and Delta 2s are working again.

**Comment by lpanger** [comment]
I was able to get parameters for my river 2 pro but when I tried with the river 3 plus it returned nothing. Has anyone else tried this?

**Reply by mjkadel** [reply]
The Ecoflow API recognizes the River 3 serial number but doesn't return any values. Ecoflow support told me they don't support that device via the API at all. I'm assuming that's why the River 3 hasn't been (and maybe can't be) added to hassio-ecoflow-cloud.

**Reply by tomaszn** [reply]
Another project collected the parameters for River 3 Plus:
https://github.com/foxthefox/ioBroker.ecoflow-mqtt/blob/main/doc/devices/river3plus.md

**Comment by Crixcel14** [comment]
First time using home assistant for the sole reason of integrating my River 3 Plus. Hoping an integration soon.

**Comment by ahwman** [comment]
Please add RIVER 3 Plus support!  Would be so appreciated… 😀

**Comment by Rogerrp1** [comment]
Since Nov 2024? Can we ask Ecoflow themselves to help solve this? Would doing a small fundraiser for River 3 integration be possible? How can we help?

**Comment by raidenGitHub** [comment]
Please login and use the URL 

https://www.ecoflow.com/us/support/get-support 

If more customers request then maybe ecoflow will assist.

<img width="1080" height="2059" alt="Screenshot_20260716_132551_Firefox~2" src="https://github.com/user-attachments/assets/abadaa90-f126-4302-9128-72a97d1e378a" />
